src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.cloud.custompolicies.app.rest;
+
+import com.redhat.cloud.custompolicies.app.NotificationSystem;
+import com.redhat.cloud.custompolicies.app.auth.RhIdPrincipal;
+import com.redhat.cloud.custompolicies.app.model.SettingsValues;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.EntityTag;
+import javax.ws.rs.core.Response;
+import org.eclipse.microprofile.metrics.annotation.Timed;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+/**
+ * @author hrupp
+ */
+@Path("/user-config")
+@Produces("application/json")
+@Consumes("application/json")
+@Timed
+@RequestScoped
+public class SettingsService {
+
+
+  @SuppressWarnings("CdiInjectionPointsInspection")
+  @Inject
+  RhIdPrincipal user;
+
+  @Inject
+  @RestClient
+  NotificationSystem notifications;
+
+  @Operation(summary = "Save or update settings from the settings UI")
+  @APIResponse(responseCode = "200", description = "Saving was ok")
+  @POST
+  @Path("/email-preference")
```

Comment:
We might need to change this so that the endpoint is `/settings` in both load/save.
Asking Karel, but already got WIP to have this working. Will send a PR to your branch later today.

Reply:
I talked to Karel and he told me I did the right thing